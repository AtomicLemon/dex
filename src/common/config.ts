import { Validator } from 'jsonschema';

import { configFile, configFileIEO, configTemplateFile } from '../config';
import { ConfigFile, ConfigFileIEO, CurrencyPairMetaData } from '../util/types';

import { configIEOSchema, configSchema, schemas } from './configSchema';
import { TokenMetaData } from './tokens_meta_data';

export class Config {
    private static _instance: Config;
    private readonly _validator: Validator;
    private _config: ConfigFile;
    public static getInstance(): Config {
        if (!Config._instance) {
            Config._instance = new Config();
        }
        return Config._instance;
    }

    public static getConfig(): ConfigFile {
        return this.getInstance()._config;
    }

    constructor() {
        this._validator = new Validator();
        for (const schema of schemas) {
            this._validator.addSchema(schema, schema.id);
        }
        this._validator.validate(configFile, configSchema, { throwError: true });
        this._config = configFile;
    }

    public _setConfig(config: ConfigFile): void {
        this._validator.validate(configFile, configSchema, { throwError: true });
        this._config = config;
    }
}

export class ConfigTemplate {
    private static _instance: ConfigTemplate;
    private readonly _validator: Validator;
    private _config: ConfigFile;
    public static getInstance(): ConfigTemplate {
        if (!ConfigTemplate._instance) {
            ConfigTemplate._instance = new ConfigTemplate();
        }
        return ConfigTemplate._instance;
    }

    public static getConfig(): ConfigFile {
        return this.getInstance()._config;
    }

    constructor() {
        this._validator = new Validator();
        for (const schema of schemas) {
            this._validator.addSchema(schema, schema.id);
        }
        this._validator.validate(configTemplateFile, configSchema, { throwError: true });
        this._config = configTemplateFile;
    }

    public _setConfig(config: ConfigFile): void {
        this._validator.validate(configTemplateFile, configSchema, { throwError: true });
        this._config = configTemplateFile;
    }

    public setPairs(pairs: CurrencyPairMetaData[]): void {
        this._config.pairs = pairs;
    }
    public setTokens(tokens: TokenMetaData[]): void {
        this._config.tokens = tokens;
    }
}

export class ConfigIEO {
    private static _instance: ConfigIEO;
    private readonly _validator: Validator;
    private readonly _config: ConfigFileIEO;
    public static getInstance(): ConfigIEO {
        if (!ConfigIEO._instance) {
            ConfigIEO._instance = new ConfigIEO();
        }
        return ConfigIEO._instance;
    }

    public static getConfig(): ConfigFileIEO {
        return this.getInstance()._config;
    }
    constructor() {
        this._validator = new Validator();
        for (const schema of schemas) {
            this._validator.addSchema(schema, schema.id);
        }
        this._validator.validate(configFileIEO, configIEOSchema, { throwError: true });
        this._config = configFileIEO;
    }
}
