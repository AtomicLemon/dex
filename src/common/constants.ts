export const RELAYER_URL = process.env.REACT_APP_RELAYER_URL || 'http://localhost:3001/api/v2';

// Must be the same a as the relayer
export const RELAYER_NETWORK_ID = process.env.REACT_APP_RELAYER_NETWORK_ID || 50;
