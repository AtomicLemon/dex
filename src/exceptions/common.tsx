export const INSUFFICIENT_MAKER_BALANCE_ERR = 'INSUFFICIENT MAKER BALANCE';
export const SIGNATURE_ERR = 'User denied message signature';
export const INSUFFICIENT_ORDERS_TO_FILL_AMOUNT_ERR = 'There are no enough orders to fill this amount';
export const RELAYER_ERR = 'There was an error with the relayer';
