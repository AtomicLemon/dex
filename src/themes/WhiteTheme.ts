import { BasicTheme } from './BasicTheme';
import { ThemeColors } from './ThemeCommons';

const whiteThemeColors: ThemeColors = {
    background: '#f5f5f5',
    borderColor: '#DEDEDE',
    darkBlue: '#002979',
    darkGray: '#474747',
    darkerGray: '#666',
    errorButtonBackground: '#FF6534',
    errorCardBackground: '#FAF4EF',
    errorCardBorder: '#F39E4B',
    errorCardText: '#F68C24',
    gray: '#808080',
    green: '#3CB34F',
    lightGray: '#B9B9B9',
    orange: '#F6851B',
    rowActive: '#FBFDFF',
    textLight: '#999',
    notificationActive: '#F8F8F8',
};

export class WhiteTheme extends BasicTheme {
    constructor() {
        super(whiteThemeColors);
    }
}
