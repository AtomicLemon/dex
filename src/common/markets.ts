import * as Config from '../config.json';

export const availableMarkets = Config.pairs;
