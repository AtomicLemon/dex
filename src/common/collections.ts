import { CollectibleCollection } from "../util/types";
import { collectibleCollectionConfig } from "../config";
import { CHAIN_ID } from "./constants";



let collectibleCollection: CollectibleCollection[] = [];

const getContractFromChainId = (addresses: Addresses) => {
    switch (CHAIN_ID) {
        case 1:
            return addresses.mainnet;
        case 3:
            return addresses.ropsten;
        case 42:
             return addresses.kovan;
        default:
           return null;
    }
}

collectibleCollection = collectibleCollectionConfig.collections.filter(c=> getContractFromChainId(c.addresses) !== null).map(c => {
    return {
        name: c.name,
        address: getContractFromChainId(c.addresses) || '0x',
        description: c.description,
        icon: c.icon,
        symbol: c.symbol || c.name.toLowerCase(),
    }
})

interface Addresses{
    mainnet?: string;
    ropsten?: string;
    kovan?: string;
}

export const getCollectibleCollections = (): CollectibleCollection[] => {
    return collectibleCollection;
}


