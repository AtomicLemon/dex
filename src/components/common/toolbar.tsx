import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { goToHome, goToWallet } from '../../store/actions';
import { getThemeColors, getWeb3State } from '../../store/selectors';
import { BasicTheme } from '../../themes/BasicTheme';
import { themeBreakPoints, themeDimensions } from '../../themes/ThemeCommons';
import { errorsWallet } from '../../util/error_messages';
import { StoreState, Web3State } from '../../util/types';
import { WalletConnectionStatusContainer } from '../account';
import { NotificationsDropdownContainer } from '../notifications/notifications_dropdown';

import { ErrorCard, ErrorIcons, FontSize } from './error_card';
import { Logo } from './logo';
import { MarketsDropdownContainer } from './markets_dropdown';

interface StateProps {
    web3State?: Web3State;
    themeColorsConfig: BasicTheme;
}

interface DispatchProps {
    onGoToHome: () => any;
    onGoToWallet: () => any;
}

type Props = StateProps & DispatchProps;

const separatorTopbar = styled.div<{ themeColors: BasicTheme }>`
    &:after {
        background-color: ${props => props.themeColors.borderColor};
        content: '';
        height: 26px;
        margin-left: 17px;
        margin-right: 17px;
        width: 1px;
    }

    &:last-child:after {
        display: none;
    }
`;

const ToolbarWrapper = styled.div<{ themeColors: BasicTheme }>`
    align-items: center;
    background: #ffffff;
    border-bottom: 1px solid ${props => props.themeColors.borderColor};
    display: flex;
    flex-grow: 0;
    flex-shrink: 0;
    height: 64px;
    justify-content: space-between;
    padding: 0 ${themeDimensions.horizontalPadding};
    position: sticky;
    z-index: 123;
`;

const MyWalletLink = styled.a`
    align-items: center;
    color: #333333;
    display: flex;
    font-size: 16px;
    font-weight: 500;
    text-decoration: none;

    &:hover {
        text-decoration: underline;
    }

    ${separatorTopbar}
`;

const ToolbarStart = styled.div`
    align-items: center;
    display: flex;
    justify-content: flex-start;
`;

const ToolbarEnd = styled.div`
    align-items: center;
    display: flex;
    justify-content: flex-end;
`;

const LogoHeader = styled(Logo)`
    ${separatorTopbar}
`;

const MarketsDropdownHeader = styled<any>(MarketsDropdownContainer)`
    align-items: center;
    display: flex;

    ${separatorTopbar}
`;

const WalletDropdown = styled(WalletConnectionStatusContainer)`
    display: none;

    @media (min-width: ${themeBreakPoints.sm}) {
        align-items: center;
        display: flex;

        ${separatorTopbar}
    }
`;

const Toolbar = (props: Props) => {
    const isMmLocked = props.web3State === Web3State.Locked;
    const isMmNotInstalled = props.web3State === Web3State.NotInstalled;
    const isMmLoading = props.web3State === Web3State.Loading;
    const { themeColorsConfig } = props;

    const handleLogoClick: React.EventHandler<React.MouseEvent> = e => {
        e.preventDefault();
        props.onGoToHome();
    };

    const handleMyWalletClick: React.EventHandler<React.MouseEvent> = e => {
        e.preventDefault();
        props.onGoToWallet();
    };

    return (
        <ToolbarWrapper themeColors={themeColorsConfig}>
            <ToolbarStart>
                <LogoHeader onClick={handleLogoClick} />
                <MarketsDropdownHeader shouldCloseDropdownBodyOnClick={false} />
            </ToolbarStart>
            {isMmLocked ? (
                <ErrorCard
                    fontSize={FontSize.Large}
                    text={errorsWallet.mmLocked}
                    icon={ErrorIcons.Lock}
                    themeColors={themeColorsConfig}
                />
            ) : null}
            {isMmNotInstalled ? (
                <ErrorCard
                    fontSize={FontSize.Large}
                    text={errorsWallet.mmNotInstalled}
                    icon={ErrorIcons.Metamask}
                    themeColors={themeColorsConfig}
                />
            ) : null}
            {isMmLoading ? (
                <ErrorCard
                    fontSize={FontSize.Large}
                    text={errorsWallet.mmLoading}
                    icon={ErrorIcons.Metamask}
                    themeColors={themeColorsConfig}
                />
            ) : null}
            {!isMmLocked && !isMmNotInstalled && !isMmLoading ? (
                <ToolbarEnd>
                    <MyWalletLink href="/my-wallet" onClick={handleMyWalletClick}>
                        My Wallet
                    </MyWalletLink>
                    <WalletDropdown themeColors={themeColorsConfig} />
                    <NotificationsDropdownContainer />
                </ToolbarEnd>
            ) : null}
        </ToolbarWrapper>
    );
};

const mapStateToProps = (state: StoreState): StateProps => {
    return {
        web3State: getWeb3State(state),
        themeColorsConfig: getThemeColors(state),
    };
};

const mapDispatchToProps = (dispatch: any): DispatchProps => {
    return {
        onGoToHome: () => dispatch(goToHome()),
        onGoToWallet: () => dispatch(goToWallet()),
    };
};

const ToolbarContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(Toolbar);

export { Toolbar, ToolbarContainer };
