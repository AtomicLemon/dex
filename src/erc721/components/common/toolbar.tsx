import React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';

import { WalletConnectionStatusContainer } from '../../../components/account';
import { ErrorCard, ErrorIcons, FontSize } from '../../../components/common/error_card';
import { NotificationsDropdownContainer } from '../../../components/notifications/notifications_dropdown';
import { goToHomeErc721, goToMyCollectibles } from '../../../store/router/actions';
import { getWeb3State } from '../../../store/selectors';
import { themeBreakPoints, themeColors, themeDimensions } from '../../../themes/commons';
import { errorsWallet } from '../../../util/error_messages';
import { StoreState, Web3State } from '../../../util/types';

import { Search } from './inputSearch';
import { Logo } from './logo';

interface StateProps {
    web3State?: Web3State;
}

interface DispatchProps {
    onGoToHome: () => any;
    goToMyCollectibles: () => any;
}

type Props = StateProps & DispatchProps;

const separatorTopbar = `
    &:after {
        background-color: ${themeColors.borderColor};
        content: "";
        height: 26px;
        margin-left: 17px;
        margin-right: 17px;
        width: 1px;
    }

    &:last-child:after {
        display: none;
    }
`;

const ToolbarWrapper = styled.div`
    align-items: center;
    background: #ffffff;
    border-bottom: 1px solid ${themeColors.borderColor};
    display: flex;
    flex-grow: 0;
    flex-shrink: 0;
    height: 64px;
    justify-content: space-between;
    padding: 0 ${themeDimensions.horizontalPadding};
    position: sticky;
    z-index: 123;
`;

const MyWalletLink = styled.a`
    align-items: center;
    color: #333333;
    display: flex;
    font-size: 16px;
    font-weight: 500;
    text-decoration: none;

    &:hover {
        text-decoration: underline;
    }

    ${separatorTopbar}
`;

const ToolbarStart = styled.div`
    align-items: center;
    display: flex;
    justify-content: flex-start;
`;

const ToolbarEnd = styled.div`
    align-items: center;
    display: flex;
    justify-content: flex-end;
`;

const LogoHeader = styled(Logo)`
    ${separatorTopbar}
`;

const WalletDropdown = styled(WalletConnectionStatusContainer)`
    display: none;

    @media (min-width: ${themeBreakPoints.sm}) {
        align-items: center;
        display: flex;

        ${separatorTopbar}
    }
`;

const Toolbar = (props: Props) => {
    const isMmLocked = props.web3State === Web3State.Locked;
    const isMmNotInstalled = props.web3State === Web3State.NotInstalled;
    const isMmLoading = props.web3State === Web3State.Loading;

    const handleLogoClick: React.EventHandler<React.MouseEvent> = e => {
        e.preventDefault();
        props.onGoToHome();
    };

    const handleMyWalletClick: React.EventHandler<React.MouseEvent> = e => {
        e.preventDefault();
        props.goToMyCollectibles();
    };

    return (
        <ToolbarWrapper>
            <ToolbarStart>
                <LogoHeader onClick={handleLogoClick} />
            </ToolbarStart>
            <Search />
            {isMmLocked ? (
                <ErrorCard fontSize={FontSize.Large} text={errorsWallet.mmLocked} icon={ErrorIcons.Lock} />
            ) : null}
            {isMmNotInstalled ? (
                <ErrorCard fontSize={FontSize.Large} text={errorsWallet.mmNotInstalled} icon={ErrorIcons.Metamask} />
            ) : null}
            {isMmLoading ? (
                <ErrorCard fontSize={FontSize.Large} text={errorsWallet.mmLoading} icon={ErrorIcons.Metamask} />
            ) : null}
            {!isMmLocked && !isMmNotInstalled && !isMmLoading ? (
                <ToolbarEnd>
                    <MyWalletLink href="/my-collectibles" onClick={handleMyWalletClick}>
                        My Collectibles
                    </MyWalletLink>
                    <WalletDropdown />
                    <NotificationsDropdownContainer />
                </ToolbarEnd>
            ) : null}
        </ToolbarWrapper>
    );
};

const mapStateToProps = (state: StoreState): StateProps => {
    return {
        web3State: getWeb3State(state),
    };
};

const mapDispatchToProps = (dispatch: any): DispatchProps => {
    return {
        onGoToHome: () => dispatch(goToHomeErc721()),
        goToMyCollectibles: () => dispatch(goToMyCollectibles()),
    };
};

const ToolbarContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(Toolbar);

export { Toolbar, ToolbarContainer };
