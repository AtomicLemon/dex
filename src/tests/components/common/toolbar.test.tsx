import { shallow } from 'enzyme';
import React from 'react';

import { Toolbar } from '../../../components/common/toolbar';
import { WhiteTheme } from '../../../themes/WhiteTheme';

describe('Toolbar', () => {
    it('Toolbar to match snapshot', () => {
        // given
        const toolbar = <Toolbar onGoToHome={jest.fn()} onGoToWallet={jest.fn()} themeColors={new WhiteTheme()} />;

        // when
        const wrapper = shallow(toolbar);

        // then
        expect(wrapper).toMatchSnapshot();
    });
});
