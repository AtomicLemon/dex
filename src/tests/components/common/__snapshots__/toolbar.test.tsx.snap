// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar Toolbar to match snapshot 1`] = `
<div
  className="sc-ifAKCX kYjrgn"
>
  <div
    className="sc-EHOje jtSruX"
  >
    <label>
      Start Content
    </label>
  </div>
  <div
    className="sc-bZQynM kEqDtR"
  >
    <label>
      End Content
    </label>
  </div>
</div>
`;
